### ISSUE TITLE
Couldn't install arachni gem : Unresolved dependency found during sorting nokogiri (>= 1.5.9)

### ISSUE THREAD

**Issue by gowthamsadasivam** [Arachni/arachni#682]
I'm trying to install Arachni in my Ubuntu 14.04 system. I was just following the install instructions from the wiki page: https://github.com/Arachni/arachni/wiki/Installation#nix-gem

while on step: `gem install arachni` I got the error :

> Unresolved dependency found during sorting - nokogiri (>= 1.5.9) (requested by loofah-2.0.3)

![arachni-2](https://cloud.githubusercontent.com/assets/5887533/13392635/f479db70-df01-11e5-8042-40c3d5e725ce.jpg)

After this error I tried installing from source (https://github.com/Arachni/arachni/wiki/Installation#nix-source) and got stuck at the same error:

![arachni-1](https://cloud.githubusercontent.com/assets/5887533/13392643/0a46d28c-df02-11e5-8499-29f227872d19.jpg)

The system default Ruby version is **Ruby 1.9.1** which is too old. So I'm using **Ruby 2.2.1** with **rvm**.
Any way to fix is error?

**Comment by Zapotek** [comment]
Hello,

I'll look into this, although is there a reason why you're not using the self-contained packages?
They're there to ensure that these issues don't affect users and also provide all necessary system dependencies.

Cheers

**Comment by gowthamsadasivam** [comment]
Thanks for the quick response. Yes, I tried using the *self-contained* packages. I had a different issue with that. The binary is executing within the package directory, but when I added it to the system $PATH and tried executing it, it had some **RVM environment** error that, I'm not sure how to fix (It would be great if you can help me with this following error).

![arachni-3](https://cloud.githubusercontent.com/assets/5887533/13394687/12bb5d94-df10-11e5-8bf9-9e78c236d69a.jpg)

**That's why I tried to build and install the gem myself.**

**Comment by Zapotek** [comment]
I think there's an Arachni gem somewhere in your path via RVM.
Please uninstall it and the package will work, right now the package is ignored.

**Comment by gowthamsadasivam** [comment]
Yes, there was an Arachni gem file in the RVM path and I just removed it. The self-contained package works perfectly now :+1: Thanks lot for the help :)

**Comment by Zapotek** [comment]
Good to know, cheers.

**Comment by anshooverma6114** [comment]
Dear All,

I am facing same issue with Ubuntu os 14 version 

## gem install arachni
Error loading RubyGems plugin "/usr/lib/ruby/gems/1.9.1/gems/rubyjams-0.1.0/lib/                                                                                        rubygems_plugin.rb": cannot load such file -- jbundler/aether (LoadError)
ERROR:  While executing gem ... (Gem::DependencyError)
    Unresolved dependency found during sorting - nokogiri (>= 1.5.9) (requested by loofah-2.0.3)

how to delete Arachni gem file in the RVM path.??

any idea???

**Comment by Zapotek** [comment]
@anshooverma6114 Is there a reason why you're trying to install the Gem rather than use the provided [packages](http://www.arachni-scanner.com/download/)?

**Comment by anshooverma6114** [comment]
@zapotek: no reason for trying with gem but I tried with apt-get install and yr scripting self contained package also . But during the package installation I get error no.5/12 step .

So how can we do that. ..???

**Comment by Zapotek** [comment]
There are no official packages you can install with `apt-get` and you don't need to use the scripts, just download one of the the provided packages linked to and you'll be good to go.

**Comment by anshooverma6114** [comment]
Ok please share the link so that I can download for ubantu 14.0.

**Comment by Zapotek** [comment]
Here you go: http://www.arachni-scanner.com/download/

**Comment by anshooverma6114** [comment]
I have already downloaded tar.gz and extract the file and run •/arachni _Web but lot of dependency asking and this is for  Linux and other os not for ubantu.  Could u please tell the exact setup.
And it this will support to ubantu den how will install the arachni please share step by step command.

**Comment by Zapotek** [comment]
Can you please show me the output of the command you ran?
Those packages already include all dependencies and there are no known issues with them.

On March 21, 2016 6:51:28 PM GMT+02:00, anshooverma6114 <notifications@github.com> wrote:
>I have already downloaded tar.gz and extract the file and run •/arachni
>_Web but lot of dependency asking and this is for  Linux and other os
>not for ubantu.  Could u please tell the exact setup.
>And it this will support to ubantu den how will install the arachni
>please share step by step command.
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/Arachni/arachni/issues/682#issuecomment-199374372

-- 
Sent from my phone, please excuse my brevity.

**Comment by anshooverma6114** [comment]
Dear I m not in office but I will show all output tomorrow morning IST time. 

I already downloaded and extract the setup and thenjoy I get some error will share you tomorrow morning.

**Comment by Joeyn414** [comment]
Im having a similar issue trying to run this in an alpine image. I would like to run gem install arachni but I am getting the same error regarding loofah asking for nokogiri >=1.5.9 however I have nokogiri 1.6.8 and 1.6.8rc2. I tried to run the package you provided for the linux build however it has other issues. Ideally I would like to run the gem within my docker alpine image. Any ideas?

**Comment by Zapotek** [comment]
@Joeyn414 The package should have worked, it this the full error you're getting?

**Comment by Joeyn414** [comment]
Thats the full error for the packages but I think it may be harder to get the packages to work rather than running it as a gem inside my alpine image. Here are the error messages from trying to install the gem:
![image](https://cloud.githubusercontent.com/assets/4161494/16451750/dcf5e7b6-3dd2-11e6-8ab8-0ebc7b13b221.png)
![image](https://cloud.githubusercontent.com/assets/4161494/16451767/f669837e-3dd2-11e6-81db-d177c3fb91dd.png)
![image](https://cloud.githubusercontent.com/assets/4161494/16451861/8117d034-3dd3-11e6-9112-14b5ae3d18c8.png)

**Comment by Zapotek** [comment]
The gem seems to be broken but the packages should work everywhere, you just need to extract them.
Can you check for the existence of `system/usr/bin/ruby` within the package please?

**Comment by Joeyn414** [comment]
Looks like its there: 
![image](https://cloud.githubusercontent.com/assets/4161494/16453114/b94aa214-3dd9-11e6-9a26-6137886a7b91.png)

**Comment by Joeyn414** [comment]
I got it to work after installing nokogiri -v 1.6.2.1 -- --use system-libraries and then fixing the fbuffer file inside json-1.8.1 gem. I then installed arachni -v 1.0.6. The dependencies for gem install arachni -v 1.0.6 for alpine are: ruby-dev, libxml2, libxml2-dev, libxslt-dev, build-base, libffi-dev, and curl. Then the gems required are io-console. 

For the fbuffer the fix is on this page: https://github.com/flori/json/issues/229 essentially modifying one of the lines.
